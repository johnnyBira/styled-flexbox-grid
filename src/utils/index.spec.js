import gridFactory, { gutterCalc, getColumnWidth } from './';

describe('Grid/utils', () => {
  describe('gutterCalc', () => {
    it('calculates the gutter from a base valuea and default multiplier', () => {
      expect(gutterCalc(24, 2)).toEqual(24);
    });
    it('calculates the gutter from a base valuea and a specified multiplier', () => {
      expect(gutterCalc(24, 2)).toEqual(24);
    });
  });
  describe('getColumnWidth', () => {
    it('returns a percentage based column width based on the given params', () => {
      expect(getColumnWidth(12, 1200, 6)).toEqual('50%');
    });
    it('returns a pixel based column width based on the given params', () => {
      expect(getColumnWidth(12, 1200, 6, true)).toEqual('600px');
    });
  });
  describe('gridFactory', () => {
    describe('default config', () => {
      const grid = gridFactory();

      it('returns an object of grid helper methods', () => {
        expect(grid).toMatchSnapshot();
      });

      it('prefills it\'s helper methods with default grid configs', () => {
        expect(grid.getColumnWidth(6)).toEqual('50%');
        expect(grid.getColumnWidth(6, true)).toEqual('600px');
        expect(grid.getGutterWidth(2)).toEqual('1rem');
      });
    });
    describe('custom config', () => {
      const grid = gridFactory({
        columnCount: 10,
        rowWidth: 1000,
        gutterBase: 24,
        gutterUnit: 'px',
      });

      it('returns an object of grid helper methods', () => {
        expect(grid).toMatchSnapshot();
      });

      it('prefills it\'s helper methods with custom grid configs', () => {
        expect(grid.getColumnWidth(5)).toEqual('50%');
        expect(grid.getColumnWidth(5, true)).toEqual('500px');
        expect(grid.getGutterWidth(2)).toEqual('24px');
      });
    });
  });
});
