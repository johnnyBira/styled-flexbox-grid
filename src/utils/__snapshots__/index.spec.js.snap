// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid/utils gridFactory custom config returns an object of grid helper methods 1`] = `
Object {
  "getColumnWidth": [Function],
  "getGutterWidth": [Function],
}
`;

exports[`Grid/utils gridFactory default config returns an object of grid helper methods 1`] = `
Object {
  "getColumnWidth": [Function],
  "getGutterWidth": [Function],
}
`;
